<?xml version="1.0" encoding="UTF-8"?>
<testsuites name="Node.js Demoapp" tests="4" time="0.000">
  <testsuite name="Check Home Page" id="5683341d-fa8c-4e02-ada0-0f61c55c42fa" timestamp="2021-02-26T16:05:19.822Z" tests="3" failures="3" errors="1" time="0.000">
    <system-err><![CDATA[Iteration: 0
RequestError: Error: getaddrinfo ENOTFOUND {{apphost}}
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:67:26)

---
]]></system-err>
    <testcase name="Home Page: Successful GET request" time="0.000" classname="NodeJsDemoapp">
      <failure type="AssertionFailure" message="expected response to have status reason 'OK' but got 'UNDEFINED'">
        <![CDATA[Failed 1 times.]]>
        <![CDATA[Collection JSON ID: 8c90c22b-3e1f-45c9-aaf5-058b37f34474.]]>
        <![CDATA[Collection name: Node.js Demoapp.]]>
        <![CDATA[Request name: Check Home Page.]]>
        <![CDATA[Test description: Home Page: Successful GET request.]]>
        <![CDATA[Error message: expected response to have status reason 'OK' but got 'UNDEFINED'.]]>
        <![CDATA[Stacktrace: AssertionError: expected response to have status reason 'OK' but got 'UNDEFINED'
   at Object.eval sandbox-script.js:1:1).]]>
      </failure>
    </testcase>
    <testcase name="Home Page: Response valid &amp; HTML body" time="0.000" classname="NodeJsDemoapp">
      <failure type="AssertionFailure" message="expected response to have content in body">
        <![CDATA[Failed 1 times.]]>
        <![CDATA[Collection JSON ID: 8c90c22b-3e1f-45c9-aaf5-058b37f34474.]]>
        <![CDATA[Collection name: Node.js Demoapp.]]>
        <![CDATA[Request name: Check Home Page.]]>
        <![CDATA[Test description: Home Page: Response valid & HTML body.]]>
        <![CDATA[Error message: expected response to have content in body.]]>
        <![CDATA[Stacktrace: AssertionError: expected response to have content in body
   at Object.eval sandbox-script.js:2:1).]]>
      </failure>
    </testcase>
    <testcase name="Home Page: Check content" time="0.000" classname="NodeJsDemoapp">
      <failure type="AssertionFailure" message="object tested must be an array, a map, an object, a set, a string, or a weakset, but undefined given">
        <![CDATA[Failed 1 times.]]>
        <![CDATA[Collection JSON ID: 8c90c22b-3e1f-45c9-aaf5-058b37f34474.]]>
        <![CDATA[Collection name: Node.js Demoapp.]]>
        <![CDATA[Request name: Check Home Page.]]>
        <![CDATA[Test description: Home Page: Check content.]]>
        <![CDATA[Error message: object tested must be an array, a map, an object, a set, a string, or a weakset, but undefined given.]]>
        <![CDATA[Stacktrace: AssertionError: object tested must be an array, a map, an object, a set, a string, or a weakset, but undefined given
   at Object.eval sandbox-script.js:3:1).]]>
      </failure>
    </testcase>
  </testsuite>
  <testsuite name="Check Info Page" id="0dd5a194-4859-49dc-a87e-255b7305c57b" timestamp="2021-02-26T16:05:19.822Z" tests="3" failures="3" errors="1" time="0.000">
    <system-err><![CDATA[Iteration: 0
RequestError: Error: getaddrinfo ENOTFOUND {{apphost}}
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:67:26)

---
]]></system-err>
    <testcase name="Info Page: Successful GET request" time="0.000" classname="NodeJsDemoapp">
      <failure type="AssertionFailure" message="expected response to have status reason 'OK' but got 'UNDEFINED'">
        <![CDATA[Failed 1 times.]]>
        <![CDATA[Collection JSON ID: 8c90c22b-3e1f-45c9-aaf5-058b37f34474.]]>
        <![CDATA[Collection name: Node.js Demoapp.]]>
        <![CDATA[Request name: Check Info Page.]]>
        <![CDATA[Test description: Info Page: Successful GET request.]]>
        <![CDATA[Error message: expected response to have status reason 'OK' but got 'UNDEFINED'.]]>
        <![CDATA[Stacktrace: AssertionError: expected response to have status reason 'OK' but got 'UNDEFINED'
   at Object.eval sandbox-script.js:1:2).]]>
      </failure>
    </testcase>
    <testcase name="Info Page: Response valid &amp; HTML body" time="0.000" classname="NodeJsDemoapp">
      <failure type="AssertionFailure" message="expected response to have content in body">
        <![CDATA[Failed 1 times.]]>
        <![CDATA[Collection JSON ID: 8c90c22b-3e1f-45c9-aaf5-058b37f34474.]]>
        <![CDATA[Collection name: Node.js Demoapp.]]>
        <![CDATA[Request name: Check Info Page.]]>
        <![CDATA[Test description: Info Page: Response valid & HTML body.]]>
        <![CDATA[Error message: expected response to have content in body.]]>
        <![CDATA[Stacktrace: AssertionError: expected response to have content in body
   at Object.eval sandbox-script.js:2:2).]]>
      </failure>
    </testcase>
    <testcase name="Info Page: Check content" time="0.000" classname="NodeJsDemoapp">
      <failure type="AssertionFailure" message="object tested must be an array, a map, an object, a set, a string, or a weakset, but undefined given">
        <![CDATA[Failed 1 times.]]>
        <![CDATA[Collection JSON ID: 8c90c22b-3e1f-45c9-aaf5-058b37f34474.]]>
        <![CDATA[Collection name: Node.js Demoapp.]]>
        <![CDATA[Request name: Check Info Page.]]>
        <![CDATA[Test description: Info Page: Check content.]]>
        <![CDATA[Error message: object tested must be an array, a map, an object, a set, a string, or a weakset, but undefined given.]]>
        <![CDATA[Stacktrace: AssertionError: object tested must be an array, a map, an object, a set, a string, or a weakset, but undefined given
   at Object.eval sandbox-script.js:3:2).]]>
      </failure>
    </testcase>
  </testsuite>
  <testsuite name="Check Weather API" id="bc94fd77-e410-40e1-895b-bed538215b3d" timestamp="2021-02-26T16:05:19.822Z" tests="3" failures="3" errors="1" time="0.000">
    <system-err><![CDATA[Iteration: 0
RequestError: Error: getaddrinfo ENOTFOUND {{apphost}}
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:67:26)

---
]]></system-err>
    <testcase name="Weather API: Successful GET request" time="0.000" classname="NodeJsDemoapp">
      <failure type="AssertionFailure" message="expected response to have status reason 'OK' but got 'UNDEFINED'">
        <![CDATA[Failed 1 times.]]>
        <![CDATA[Collection JSON ID: 8c90c22b-3e1f-45c9-aaf5-058b37f34474.]]>
        <![CDATA[Collection name: Node.js Demoapp.]]>
        <![CDATA[Request name: Check Weather API.]]>
        <![CDATA[Test description: Weather API: Successful GET request.]]>
        <![CDATA[Error message: expected response to have status reason 'OK' but got 'UNDEFINED'.]]>
        <![CDATA[Stacktrace: AssertionError: expected response to have status reason 'OK' but got 'UNDEFINED'
   at Object.eval sandbox-script.js:1:3).]]>
      </failure>
    </testcase>
    <testcase name="Weather API: Response valid &amp; JSON body" time="0.000" classname="NodeJsDemoapp">
      <failure type="AssertionFailure" message="expected response to have content in body">
        <![CDATA[Failed 1 times.]]>
        <![CDATA[Collection JSON ID: 8c90c22b-3e1f-45c9-aaf5-058b37f34474.]]>
        <![CDATA[Collection name: Node.js Demoapp.]]>
        <![CDATA[Request name: Check Weather API.]]>
        <![CDATA[Test description: Weather API: Response valid & JSON body.]]>
        <![CDATA[Error message: expected response to have content in body.]]>
        <![CDATA[Stacktrace: AssertionError: expected response to have content in body
   at Object.eval sandbox-script.js:2:3).]]>
      </failure>
    </testcase>
    <testcase name="Weather API: Check API response" time="0.000" classname="NodeJsDemoapp">
      <failure type="AssertionFailure" message="Unexpected token u in JSON at position 0">
        <![CDATA[Failed 1 times.]]>
        <![CDATA[Collection JSON ID: 8c90c22b-3e1f-45c9-aaf5-058b37f34474.]]>
        <![CDATA[Collection name: Node.js Demoapp.]]>
        <![CDATA[Request name: Check Weather API.]]>
        <![CDATA[Test description: Weather API: Check API response.]]>
        <![CDATA[Error message: Unexpected token u in JSON at position 0.]]>
        <![CDATA[Stacktrace: JSONError: Unexpected token u in JSON at position 0
   at Object.eval sandbox-script.js:3:3).]]>
      </failure>
    </testcase>
  </testsuite>
  <testsuite name="Check Error Page" id="11f9fe5b-4164-4e6f-9174-c969195d41f3" timestamp="2021-02-26T16:05:19.822Z" tests="3" failures="3" errors="1" time="0.000">
    <system-err><![CDATA[Iteration: 0
RequestError: Error: getaddrinfo ENOTFOUND {{apphost}}
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:67:26)

---
]]></system-err>
    <testcase name="Error Page: Returns 500" time="0.000" classname="NodeJsDemoapp">
      <failure type="AssertionFailure" message="expected { Object (id, _details, ...) } to have property 'code'">
        <![CDATA[Failed 1 times.]]>
        <![CDATA[Collection JSON ID: 8c90c22b-3e1f-45c9-aaf5-058b37f34474.]]>
        <![CDATA[Collection name: Node.js Demoapp.]]>
        <![CDATA[Request name: Check Error Page.]]>
        <![CDATA[Test description: Error Page: Returns 500.]]>
        <![CDATA[Error message: expected { Object (id, _details, ...) } to have property 'code'.]]>
        <![CDATA[Stacktrace: AssertionError: expected { Object (id, _details, ...) } to have property 'code'
   at Object.eval sandbox-script.js:1:4).]]>
      </failure>
    </testcase>
    <testcase name="Error Page: Response valid &amp; HTML body" time="0.000" classname="NodeJsDemoapp">
      <failure type="AssertionFailure" message="expected response to have content in body">
        <![CDATA[Failed 1 times.]]>
        <![CDATA[Collection JSON ID: 8c90c22b-3e1f-45c9-aaf5-058b37f34474.]]>
        <![CDATA[Collection name: Node.js Demoapp.]]>
        <![CDATA[Request name: Check Error Page.]]>
        <![CDATA[Test description: Error Page: Response valid & HTML body.]]>
        <![CDATA[Error message: expected response to have content in body.]]>
        <![CDATA[Stacktrace: AssertionError: expected response to have content in body
   at Object.eval sandbox-script.js:2:4).]]>
      </failure>
    </testcase>
    <testcase name="Error Page: Check exception" time="0.000" classname="NodeJsDemoapp">
      <failure type="AssertionFailure" message="object tested must be an array, a map, an object, a set, a string, or a weakset, but undefined given">
        <![CDATA[Failed 1 times.]]>
        <![CDATA[Collection JSON ID: 8c90c22b-3e1f-45c9-aaf5-058b37f34474.]]>
        <![CDATA[Collection name: Node.js Demoapp.]]>
        <![CDATA[Request name: Check Error Page.]]>
        <![CDATA[Test description: Error Page: Check exception.]]>
        <![CDATA[Error message: object tested must be an array, a map, an object, a set, a string, or a weakset, but undefined given.]]>
        <![CDATA[Stacktrace: AssertionError: object tested must be an array, a map, an object, a set, a string, or a weakset, but undefined given
   at Object.eval sandbox-script.js:3:4).]]>
      </failure>
    </testcase>
  </testsuite>
</testsuites>